reasoners/base.py

```diff
@@ -326,4 +328,22 @@ def evaluate_sc(self,
     def eval_output(self, answer, output):
         pass
 
-    
\ No newline at end of file
+class ToolModule():
+    def __init__(self, func, name, description):
+        self.func = func
+        self.name = name
+        self.description = description
+        self.call_params = inspect.signature(func).parameters
+        self.tool = StructuredTool.from_function(
```

Comment:
This doesn't seem to be a desired way to integrate langchain.. What's the benefit of using StructuredTool as a member variable here?